test/playlist.test.js

```diff
@@ -1317,4 +1317,95 @@ QUnit.module('Playlist', function() {
       'no segments, live delay can\'t be calculated'
     );
   });
+
+  QUnit.test('playlistMatch', function(assert) {
+    assert.equal(Playlist.playlistMatch(null, null), false, 'null playlists do not match');
+    assert.equal(Playlist.playlistMatch({}, null), false, 'a playlist without b');
+    assert.equal(Playlist.playlistMatch(null, {}), false, 'b playlist without a');
+
+    const a = {id: 'foo', uri: 'foo.m3u8', resolvedUri: 'http://example.com/foo.m3u8'};
+    const b = {id: 'foo', uri: 'foo.m3u8', resolvedUri: 'http://example.com/foo.m3u8'};
+
+    assert.equal(Playlist.playlistMatch(a, a), true, 'object signature match');
+
+    assert.equal(Playlist.playlistMatch(a, b), true, 'id match');
+
+    a.id = 'bar';
+    assert.equal(Playlist.playlistMatch(a, b), true, 'resolved uri match');
+
+    a.resolvedUri += '?nope';
+    assert.equal(Playlist.playlistMatch(a, b), true, 'uri match');
+
+    a.uri += '?nope';
+
+    assert.equal(Playlist.playlistMatch(a, b), false, 'no match');
```

Comment:
should query parameters result in a uri match?

Reply:
I see, resolveUri query param succeeds because the uri still matches but if both have a query param it fails.